src/server/pages/revisions-page.jsx

```diff
@@ -70,8 +70,10 @@ export default class RevisionsPage extends React.Component {
   }
 
   onUploadFile(newFileInfo) {
-    const { files } = this.state
+    let { files } = this.state
+    files = files.filter(f => f.filename !== newFileInfo.filename)
```

Comment:
I think it's preferable not to modify the state directly (via `let`). I would do this instead:

```js
let files = this.state.files.filter(...)
...
this.setState({ files })
```